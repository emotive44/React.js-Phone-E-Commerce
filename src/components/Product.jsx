import React from 'react';

class Product extends React.Component {
    render() {
        return(
            <div>
                product
            </div>
        )
    }
}

export default Product;