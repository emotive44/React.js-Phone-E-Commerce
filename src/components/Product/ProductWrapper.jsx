import styled from 'styled-components';

const ProductWrapper = styled.div`
    .card {
        border-color: transparent;
        transition: all 0.8s linear;
    }
    .card-footer {
        background: transparent;
        transition: all 0.8s linear;
    }
    &:hover {
        .card {
            border: 0.04rem solid rgba(0,0,0,0.2);
            box-shadow: 2px 5px 10px 0px rgba(0,0,0,0.2);
        }
        .card-footer { 
            background: rgba(247, 247, 247);
        }
    }

    .img-container {
        position: relative;
        overflow: hidden;
    }
    .img-container:hover .card-img-top {
        transform: scale(1.2);
        transition: all 0.8s linear;
    }

    .cart-btn {
        position: absolute;
        bottom: 0;
        right: 0;
        border: none;
        padding: 0.2rem 0.4rem;
        background: var(--lightBlue);
        color: var(--mainWhite);
        border-radius: 0.5rem 0 0 0;
        font-size 1.5rem;
        transform: translate(100%, 100%);
        transition: all 0.8s linear;
    }
    .img-container:hover .cart-btn {
        transform: translate(0,0);
        outline: none;
    }
    .cart-btn:hover {
        color: var(--mainBlue);
    }
`;

export default ProductWrapper;
