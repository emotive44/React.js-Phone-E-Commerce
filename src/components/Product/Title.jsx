import React from 'react';

const Title = ({ name, title }) => {
    if(!name || !title) { return null };
    return (
        <div className="row">
            <div className="col-10 mx-auto my-2 text-center">
                <h1 className="text-capitalize font-weight-bold">{name}</h1>
                <h3 className="text-capitalize font-weight-bold">{title}</h3>
            </div>
        </div>
    )
}

export default Title;
