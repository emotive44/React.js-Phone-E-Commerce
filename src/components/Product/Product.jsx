import React from 'react';
import { Link } from 'react-router-dom';
import { ProductContext } from '../../context';
import ProductWrapper from './ProductWrapper';

class Product extends React.Component {
    render() {
        const { id, title, img, price, inCart } = this.props;
        return(
            <ProductWrapper className="col-9 mx-auto col-md-6 col-lg-3 my-3">
                <div className="card">
                    <ProductContext.Consumer>
                        {
                            data => (
                                <div className="img-container p-5" onClick={() => data.handleDetail(id)}>
                                    <Link to='/details'>
                                        <img src={img} alt="phone" className="card-img-top mb-3"/>
                                    </Link>
                                    
                                    <button className="cart-btn" 
                                        disabled={inCart ? true : false} 
                                        onClick={() => {data.addToCart(id); data.openModal(id) }}
                                    >
                                        {inCart
                                            ? (<p className="text-capitalize mb-0" disabled>in inCart</p>)
                                            : (<i className="fas fa-cart-plus"/>)
                                        }
                                    </button>
                                </div>
                            )
                        }
                    </ProductContext.Consumer>
                    <div className="card-footer d-flex justify-content-between">
                        <p className="align-self-center mb-0">{title}</p>
                        <h5 className="text-blue font-italic mb-0">${price}</h5>
                    </div>
                </div>
            </ProductWrapper>
        )
    }
}

export default Product;