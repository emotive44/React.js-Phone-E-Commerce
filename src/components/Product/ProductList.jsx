import React, { Fragment } from 'react';
import Product from './Product';
import Title from './Title';
import { ProductConsumer } from '../../context'

class ProductList extends React.Component {
    render() {
        return(
            <Fragment>
                <div className="py-5">
                    <div className="container">
                        <Title name="our" title="products"/>
                        <div className="row">
                            <ProductConsumer>
                                {(data) => {
                                    return data.products.map( product => {
                                        return <Product key={product.id} {...product} />
                                    });
                                }}
                            </ProductConsumer>
                        </div>
                    </div>
                </div>
            </Fragment>
        )
    }
}

export default ProductList;