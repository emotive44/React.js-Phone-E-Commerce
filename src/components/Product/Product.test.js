import React from 'react';
import Product from './Product';
import renderer from 'react-test-renderer';

jest.mock('react-router-dom', () => ({
    Link: 'Link'
}));

describe('Product Component', () => {
    test('should render return null if props is undefiend', () => {
        const component = renderer.create(<Product />);
        expect(component.toJSON()).toMatchSnapshot();
    })
})