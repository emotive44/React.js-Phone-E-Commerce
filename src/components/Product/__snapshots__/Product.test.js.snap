// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Component should render return null if props is undefiend 1`] = `
<div
  className="sc-AxjAm gmElYc col-9 mx-auto col-md-6 col-lg-3 my-3"
>
  <div
    className="card"
  >
    <div
      className="img-container p-5"
      onClick={[Function]}
    >
      <Link
        to="/details"
      >
        <img
          alt="phone"
          className="card-img-top mb-3"
        />
      </Link>
      <button
        className="cart-btn"
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fas fa-cart-plus"
        />
      </button>
    </div>
    <div
      className="card-footer d-flex justify-content-between"
    >
      <p
        className="align-self-center mb-0"
      />
      <h5
        className="text-blue font-italic mb-0"
      >
        $
      </h5>
    </div>
  </div>
</div>
`;
